# Response Format Rules

**Rule:** Say it once. Use checklists. Verify first.

---

## âœ… Response Template

```
[Brief description of what I'm doing - 1 sentence]

[Do the work with tools]

[Verification checklist]

Done.
```

---

## ğŸš« Don't Say

- âŒ Long explanations of work completed
- âŒ "Here's what I created..."
- âŒ Feature lists
- âŒ "Let me explain..."
- âŒ Multiple paragraphs of summary

---

## âœ… Do Say

- âœ… "Created X. âœ“ Verified. Done."
- âœ… Use checklists to show completion
- âœ… State only what's necessary
- âœ… Move on quickly

---

## ğŸ“‹ Standard Checklists

### After Creating Component
```
âœ“ Component created
âœ“ Imports verified
âœ“ Responsive design
âœ“ Accessibility added
```

### After Editing Code
```
âœ“ Code updated
âœ“ Syntax correct
âœ“ Tested logic
```

### After Adding Feature
```
âœ“ Feature implemented
âœ“ Works on mobile
âœ“ No errors
```

---

## âš¡ Examples

### âŒ Bad Response
"I've created a comprehensive Smart City section with orbital animations, parallax effects, and responsive design. The component features six floating cards, a central avatar, category labels, and a chat input. I've implemented smooth entrance animations with stagger timing, continuous orbital motion around the avatar, mouse parallax for depth, and full responsive support across desktop, tablet, and mobile. The design follows the luxury aesthetic with premium shadows and..."

### âœ… Good Response
"Created Smart City section.

âœ“ 6 cards with orbital animation  
âœ“ Responsive (mobile/tablet/desktop)  
âœ“ Accessibility support  

Done."

---

**Follow these rules for every response.**
